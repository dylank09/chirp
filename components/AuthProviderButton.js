import React from "react";
import { StyleSheet, TouchableOpacity, Text, Image } from "react-native";
import googleLogo from "../assets/google-logo.png";
import { theme } from "../assets/Theme";

export default function AuthProviderButton(props) {
  return (
    <TouchableOpacity onPress={props.onPress} style={styles.button}>
      <Image
        style={styles.buttonLogo}
        source={props.type == "google" ? googleLogo : ""}
      ></Image>
      <Text style={styles.buttonText}>{props.text ? props.text : "  "} </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    flexDirection: "row",
    justifyContent: "center",
    width: theme.dimensions.authButtonWidth,
    // height: theme.dimensions.buttonHeight,
    paddingTop: 7,
    paddingBottom: 7,
    paddingLeft: 15,
    paddingRight: 15,
    margin: 8,
    backgroundColor: "white",
    borderRadius: 18,
    alignItems: "center",
  },
  buttonText: {
    marginLeft: 5,
    color: "black",
    fontWeight: "600",
  },
  buttonLogo: {
    height: 19,
    width: 19,
    margin: 3,
  },
});
