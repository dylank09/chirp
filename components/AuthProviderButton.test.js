import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";

import AuthProviderButton from "./AuthProviderButton";

describe("<AuthProviderButton />", () => {
  const tree = renderer.create(<AuthProviderButton />).toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("button is present", () => {
    const { getByTestId } = render(
      <AuthProviderButton
        onPress={() => console.log("pressed")}
        type="google"
        text="Testing Button"
      />
    );
    const foundButton = getByTestId("button");
    expect(foundButton).toBeTruthy();
  });
});
