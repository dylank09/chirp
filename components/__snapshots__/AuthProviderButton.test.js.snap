// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthProviderButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 18,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 8,
      "opacity": 1,
      "paddingBottom": 7,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 7,
      "width": "70%",
    }
  }
  testID="button"
>
  <Image
    source=""
    style={
      Object {
        "height": 19,
        "margin": 3,
        "width": 19,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontWeight": "600",
        "marginLeft": 5,
      }
    }
  >
      
     
  </Text>
</View>
`;
